core/storage/statistics/gae_models_test.py

```diff
@@ -43,16 +46,189 @@ def test_create_and_get_event_models(self):
         self.assertEqual(event_model.is_feedback_useful, True)
 
 
+class StateCounterModelTests(test_utils.GenericTestBase):
+
+    def test_stat_model_gets_created(self):
+        model_instance = stats_models.StateCounterModel.get_or_create(
+            'exp_id1', 'state_name')
+        model_instance.first_entry_count = 1
```

Comment:
If you're testing the _or_create part, you don't need to set stuff here. You should instead be inserting that model_instance contains the expected values.

Reply:
The get_or_create() does not save the model(does not do put()). This is why I set these stuff so that I can assert the model_instance with the expected values. Without setting up these values there is nothing to assert except the id(which I also have asserted)

Reply:
It does not do put(), yes, but you can still assert the properties. The model still should have the necessary fields.

Reply:
Done